src/server/snapshot/snapshotter.ts

```diff
@@ -88,28 +92,42 @@ export class Snapshotter {
     });
     const initScript = '(' + frameSnapshotStreamer.toString() + ')()';
     await this._context._doAddInitScript(initScript);
-    for (const page of this._context.pages())
-      await page.mainFrame()._evaluateExpression(initScript, false, undefined, 'main');
+    const promises = [];
+    for (const page of this._context.pages()) {
+      for (const frame of page.frames())
+        promises.push(frame._evaluateExpression(initScript, false, undefined, 'main'));
+    }
+    await Promise.all(promises);
   }
 
   dispose() {
     helper.removeEventListeners(this._eventListeners);
   }
 
-  async forceSnapshot(page: Page, snapshotId: string) {
-    await Promise.all([
-      page.frames().forEach(async frame => {
-        try {
-          const context = await frame._mainContext();
-          await context.evaluateInternal(({ kSnapshotStreamer, snapshotId }) => {
-            // Do not block action execution on the actual snapshot.
-            Promise.resolve().then(() => (window as any)[kSnapshotStreamer].forceSnapshot(snapshotId));
-            return undefined;
-          }, { kSnapshotStreamer, snapshotId });
-        } catch (e) {
-        }
-      })
-    ]);
+  async captureSnapshot(page: Page, snapshotId: string) {
+    const snapshotFrame = async (frame: Frame) => {
+      const context = await frame._mainContext();
+      await context.evaluateInternal(({ kSnapshotStreamer, snapshotId }) => {
+        (window as any)[kSnapshotStreamer].captureSnapshot(snapshotId);
+        return undefined;
+      }, { kSnapshotStreamer, snapshotId });
+    };
+    await Promise.all(page.frames().map(frame => snapshotFrame(frame).catch(() => {})));
+  }
+
+  async setAutoSnapshotInterval(interval: number): Promise<void> {
```

Comment:
How do we pass this interval to any new frames attached after setAutoSnapshotInterval() call?